### PR TITLE
[Feature] Adds previous URL to support ticket

### LINKED ISSUE #4191: ✨ Support Page - Submit previous URL with support ticket
We want to submit the most recent url the user was on along with the support ticket. This may help clarify where they ran into trouble. This will involve:
1. Working with the helpdesk team to arrange a custom field
2. Building a way to know which url a user came from, before visiting the support page.

For problem 2, the first thing that occurs to me is to use the `state` function of the [History API](https://github.com/remix-run/history/blob/main/docs/api-reference.md). Any links which point to the support page could push the current location into the state, eg at the key `prevLocation`. Our form could check if such a value exists in the state, and include it in the submission if possible. 

I don't know if this is the best solution? The only other place we're using state, when moving between the Search and Request forms, feels janky and confusing to me. Open to alternatives.

## Acceptance Criteria:
- [ ] Capture the URL of the last page the user visited before support
- [ ] Send that URL to the helpdesk in a custom field

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #4191.

## 👋 Introduction

This PR adds the previous URL to a new support ticket via to the `SupportForm` component.

## 🕵️ Details

The previous URL use either the `state.referrer` value passed by the react-router `Link` component for **/support** links or the `document.referrer` for when the previous URL was outside of the single-page application (e.g., **/privacy-notice** which is part of tc-report).

## 🎩 Accolades

- @esizer 
- @marc-donofrio 

## ⚒️ To-do

```[tasklist]
### Tasks
- [x] Confirm Freshdesk ticket field with @marc-donofrio
``` 

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

> [!NOTE]  
> To test creating a support ticket, you will need to access to and [credentials for Freshdesk](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/api/config/freshdesk.php).

1. Navigate to a page outside the SPA (e.g., **/search**)
2. Click the **Contact Us** link in the footer
3. Fill out the support form and submit
4. Ensure all parts of the URL (including any params) are passed to the previous URL value in the created support ticket
5. Navigate to a page within the SPA (e.g., **/privacy-notice**)
6. Click the **Contact Us** link in the footer
7. Fill out the support form and submit
8. Ensure all parts of the URL (including any params) are passed to the previous URL value in the created support ticket
9. Repeat process for other `Link` component for **/support** links

### CODE REVIEW

File: apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx [hunk 1]
```diff
@@ -102,13 +104,16 @@ const SupportForm = ({
   currentUser,
 }: SupportFormProps) => {
   const intl = useIntl();
+  const location = useLocation();
+  const previousUrl = location?.state.referrer ?? document.referrer;
```
Comment (tristan-orourke) [inline apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx:108]: I'm getting an error at this line when I go from `/en/privacy-notice` to Contact Us page

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
I'm getting an error when going from non-spa to Contact Us page :(

File: apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx (hunk 1 above)
Comment (mnigh) [reply apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx:108]: Strange, I don't see any error. Maybe your environment needs rebuilding from switching branches?

Try adding `console.log(previousUrl);` on line 109 after the `previousUrl` is defined and building again `npm run dev`. Does anything come back in your browser console?

Comment (mnigh) [reply apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx]: Good catch @tristan-orourke, my bad. Should be fixed in 51dd01517fa80c34fa8f060bfdebd0b5c9bce049.

**Review by tristan-orourke — APPROVED** [review]
Looks good now!

**Pull request merged**